src/algorithms/utils/TimeSeries.test.ts

```diff
@@ -1,6 +1,106 @@
 import { uniformDatesBetween, makeTimeSeries, updateTimeSeries } from './TimeSeries'
 import { interpolateTimeSeries } from '../run'
 
+import * as fs from 'fs'
+const { utils } = require('jest-snapshot')
+
+expect.extend({
+  toBeCloseToArraySnapshot(received: number[]) {
+    const { testPath, currentTestName, snapshotState } = this
+    snapshotState._counters.set(currentTestName, (snapshotState._counters.get(currentTestName) || 0) + 1)
+    const count = Number(snapshotState._counters.get(currentTestName))
+    const key = utils.testNameToKey(currentTestName, count)
+    const expected = snapshotState._snapshotData[key]
+
+    /* If this isn't done, Jest reports the test as 'obsolete' and prompts for deletion. */
+    snapshotState._uncheckedKeys.delete(key)
+
+    /* If the snapshot isn't JSON, then return an empty array. Otherwise
+     * the test will fails in an inscrutable way.
+     */
+    const tryParse = (str: string) => {
+      try {
+        return JSON.parse(str)
+      } catch (e) {
+        return []
+      }
+    }
+
+    /* If the expected field is blank, then this is probably a new snapshot. */
+    const expectedDeserialized = expected ? tryParse(expected) : []
+
+    const pass =
+      expectedDeserialized.length > 0 /* must check this because every() returns true if empty array */
+        ? received.every((_, idx) => Math.abs(expectedDeserialized[idx] - received[idx]) < 10 ** -2 / 2)
+        : false
```

Comment:
Loop through the expected results and compare each float to the received results. The comparison function is equivalent to the default toBeCloseTo(), which checks for a difference of < 0.005.

Reply:
@abrie Can this be parameterized?

This is a very silly way of doing comparison, because it assumes a particular scale of values.
For example if my expected floats are `0.001` (e.g. 0.1% mortality rate in a certain age group)  and received is `0.003` (0.3%) this will fail to detect it (false negative). Similarly, false positives for large values, but that is a concern for astrophysicists, not us.

I understand that this is how Jest does things. But what do I say our hospital administrators when their patient's mortality suddenly increases 3 times? That facebookz codez are cool and everybody just needs to chill? :)

Reply:
Yes, this can be parameterized – like Jest's [`toBeCloseTo()`](https://jestjs.io/docs/en/expect#tobeclosetonumber-numdigits), which takes an optional second parameter 'numDigits`. Definitely something that should be incorporated here.